charger/ocpp.go

```diff
@@ -160,21 +157,23 @@ func NewOCPP(id string, connector int, idtag string,
 		return nil, fmt.Errorf("invalid connector: %d", connector)
 	}
 
-	conn, err := ocpp.NewConnector(log, connector, cp, timeout)
-	if err != nil {
-		return nil, err
+	if remoteStart && idtag == "" {
```

Comment:
Beim zweiten drüber nachdenken: auch remotestart braucht es egtl. nicht mehr. Optionen:

1. remotestart entfällt und gilt falls idtag != ""
2. remotestart wird ein string und idtag entfällt (das könnten wir im Template machen)
3. am ähnlichsten zu heute: remotestart bleibt- dann müsste das cp.IdTag raus. Das wäre am Einfachsten.

Aktuell gibts zwei Optionen die das Gleiche tun.

Reply:
Achso: cp.IdTag würde ich nicht nehmen. Wenn es remotestart braucht, dann muss auch deterministisch sein mit welchem Tag das passiert- das muss dann das aus der Config sein, nicht ein beliebiges vorher durch Zufall verwendetes. Das kann kein Anwender vorhersehen.

Reply:
`cp.IdTag` ist das ggf. via `getConfiguration` ermittelte und vorgegebene Tag welches bei einigen Boxen (z.B. Alfen) für Remote Transaktionen benötigt wird.
Dieses ersetzt (wenn vorhanden) automatisch das `defaultIdTag` "evcc".
Dieser gesamte Mechanismus wiederum kann jedoch immer von einem manuellen idtag-Eintrag in der Konfig überstimmt werden.

All dies ist jedoch nur relevant wenn `remoteStart` aktiv ist und wird dann intern als `remoteIdTag` im Connector verarbeitet.
Dieser Wert ist mindestens am Connector _fest_. Eigentlich sogar am CP.

Die ganze _dynamische_ Geschichte kommt nur aus der anderen Richtung: Von der lokalen Authentifizierung (RFID-Tag) wenn `remoteStart` nicht aktiv ist. Dies bekommt der Connector im StartTransaction-Callback übergeben und hält dies für das Identify-API vor.